aldryn_faq/cms_toolbar.py

```diff
@@ -89,23 +90,29 @@ def __get_faq_config(self):
 
         return config
 
-    def get_on_delete_redirect_url(self, obj):
+    def get_category_list_url(self, lang=None):
+        """
+        Returns url for faq-category-list view with respect to lang or
+        current_lang
+        """
+        with force_language(lang if lang else self.current_lang):
+            url = reverse('{0}:faq-category-list'.format(
+                self.config.namespace))
+        return url
+
+    def get_on_delete_redirect_url(self, obj, lang=None):
         if not self.config:
             self.config = self.__get_faq_config()
 
         if isinstance(obj, Category):
-            url = reverse(
-                '{0}:faq-category-list'.format(self.config.namespace)
-            )
+            url = self.get_category_list_url(lang)
         elif isinstance(obj, Question):
             category = obj.category
-            lang = obj.get_current_language()
```

Comment:
The thing is that if toolbar language is changed - it picks up the wrong language. so it has to be replaced. Not sure if that is due to changes in cms or was there from the beginning...